ui/src/app/pages/analysis/interactive-notebook.tsx

```diff
@@ -122,9 +122,14 @@ export const InteractiveNotebook = fp.flow(withUrlParams(), withCurrentWorkspace
       this.aborter.abort();
     }
 
+    clearAndSetTimeout(f, millis) {
```

Comment:
I like it. I'd also pass in the timer to be set and cleared so it's more generally usable.

Or make this something you can install on a timer class/object of some sort, so you can do `MyTimer.reset(fun, callback, duration)`. If there's a way to map functions to timers, that would cut things down even more (unless you needed simultaneous `runCluster` timers.

Whatever patterns you decide on, it'd be worth building a little library or set of components for doing all our timed events so we can tune it in one place.